v2/internal/testutils/integration.go

```diff
@@ -113,3 +118,35 @@ func NewTCPServer(handler func(conn net.Conn), port ...int) *TCPServer {
 func (s *TCPServer) Close() {
 	s.listener.Close()
 }
+
+// NewWebsocketServer creates a new websocket server from a handler
+func NewWebsocketServer(path string, handler func(conn net.Conn), originValidate func(origin string) bool, port ...int) *httptest.Server {
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if value := r.Header.Get("Origin"); value != "" && !originValidate(value) {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		conn, _, _, err := ws.UpgradeHTTP(r, w)
+		if err != nil {
+			return
```

Comment:
We should at least log the error

Reply:
It's part of the integration test so it's not really that relevant as the test will fail anyway if the functionality doesn't work as expected.

Reply:
That's true, but if the test fails, having the reason in the logs helps solving the issue faster